packages/@react-aria/disclosure/docs/useDisclosure.mdx

```diff
@@ -0,0 +1,333 @@
+
+```tsx example export=true
+import {useDisclosureState} from '@react-stately/disclosure';
+import {useDisclosure} from '@react-aria/disclosure';
+import {useButton} from '@react-aria/button';
+
+function Disclosure(props) {
+  let state = useDisclosureState(props);
+  let panelRef = React.useRef<HTMLDivElement | null>(null);
+  let triggerRef = React.useRef<HTMLButtonElement | null>(null);
+  let {buttonProps: triggerProps, panelProps} = useDisclosure(props, state, panelRef);
+  let {buttonProps} = useButton(triggerProps, triggerRef);
+
+  return (
+    <div className="disclosure">
+      <h3>
+        <button className="trigger" {...buttonProps}>
+          <svg viewBox="0 0 24 24">
+            <path d="m8.25 4.5 7.5 7.5-7.5 7.5" />
+          </svg>
+          {props.title}
+        </button>
+      </h3>
+      <div className="panel" ref={panelRef} {...panelProps}>
+        <p>
+          {props.children}
+        </p>
+      </div>
+    </div>
+  );
+};
+
+<Disclosure title="System Requirements">
+  Details about system requirements here.
+</Disclosure>
+```
+
+<details>
+  <summary style={{fontWeight: 'bold'}}><ChevronRight size="S" /> Show CSS</summary>
+
+```css
+.disclosure {
+  .trigger {
+    background: none;
+    border: none;
+    box-shadow: none;
+    font-weight: bold;
+    font-size: 16px;
+    display: flex;
+    align-items: center;
+    gap: 8px;
+
+    svg {
+      rotate: 0deg;
+      transition: rotate 200ms;
+      width: 12px;
+      height: 12px;
+      fill: none;
+      stroke: currentColor;
+      stroke-width: 3px;
+    }
+
+    &[aria-expanded="true"] svg {
+      rotate: 90deg;
+    }
+  }
+}
+
+.panel {
+  margin-left: 32px;
+}
+```
+
+</details>
+
+## Usage
+
+The following examples show how to use the `Disclosure` component created in the above example.
+
+### Default expansion
+
+Whether or not the disclosure is expanded or not by default can be set with the `defaultExpanded` prop.
+
+```tsx example
+<Disclosure title="System Requirements" defaultExpanded>
+  Details about system requirements here.
+</Disclosure>
+```
+
+### Controlled expansion
+
+Expansion can be controlled using the `isExpanded` prop, paired with the `onExpandedChange` event. The `onExpandedChange` event is fired when the user presses the trigger button.
+
+```tsx example
+function ControlledDisclosure(props) {
+  let [isExpanded, setExpanded] = React.useState(false);
+
+  return (
+    <Disclosure title="System Requirements" isExpanded={isExpanded} onExpandedChange={setExpanded}>
+      Details about system requirements here.
+    </Disclosure>
+  );
+}
+```
+
+### Disabled
+
+A disclosure can be disabled with the `isDisabled` prop. This will disable the trigger button and prevent the panel from being opened or closed.
+
+```tsx example
+<Disclosure title="System Requirements" isDisabled>
+  Details about system requirements here.
+</Disclosure>
+```
+
+## Disclosure Group
```

Comment:
feel free to give your opinion on this but i wonder if this section should just go in the `useDisclosureGroupState` docs under `Example` and then we could link to that page in the `Advanced Customization` section of DisclosureGroup? and we also link the `useDisclosure` docs in the `useDisclosureGroupState` docs (sorry if that's confusing, i'm happy to clarify)

i also understand that we don't have a `useDisclosureGroup` hook where i would guess this section would normally go in so i think this is still a reasonable place to put it if we do keep it this way

Reply:
I like that idea. It also means we can name it Disclosure and not have two on the same page.